mono/mini/mini.c

```diff
@@ -849,13 +849,18 @@ void
 mono_add_ins_to_end (MonoBasicBlock *bb, MonoInst *inst)
 {
 	int opcode;
+	MonoInst *last_ins;
 
 	if (!bb->code) {
 		MONO_ADD_INS (bb, inst);
 		return;
 	}
 
-	switch (bb->last_ins->opcode) {
+	last_ins = bb->last_ins;
+	while (last_ins->opcode == OP_SEQ_POINT || last_ins->opcode == OP_IL_SEQ_POINT)
+		last_ins = last_ins->prev;
+
```

Comment:
There is a mono_bb_last_ins () function in mini.h for this.

Also, it would be better if we wouldn't insert seq points after the branch, but thats a separate issue.

Looks ok otherwise.